sdk/openai/Azure.AI.OpenAI/src/Custom/CompletionsOptions.Serialization.cs

```diff
@@ -0,0 +1,353 @@
+                    }
+                    logprobs = property.Value.GetInt32();
+                    continue;
+                }
+                if (property.NameEquals("echo"u8))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        echo = null;
+                        continue;
+                    }
+                    echo = property.Value.GetBoolean();
+                    continue;
+                }
+                if (property.NameEquals("stop"u8))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        property.ThrowNonNullablePropertyIsNull();
+                        continue;
+                    }
+                    List<string> array = new List<string>();
+                    foreach (var item in property.Value.EnumerateArray())
+                    {
+                        array.Add(item.GetString());
+                    }
+                    stop = array;
+                    continue;
+                }
+                if (property.NameEquals("presence_penalty"u8))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        presencePenalty = null;
+                        continue;
+                    }
+                    presencePenalty = property.Value.GetSingle();
+                    continue;
+                }
+                if (property.NameEquals("frequency_penalty"u8))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        frequencyPenalty = null;
+                        continue;
+                    }
+                    frequencyPenalty = property.Value.GetSingle();
+                    continue;
+                }
+                if (property.NameEquals("best_of"u8))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        bestOf = null;
+                        continue;
+                    }
+                    bestOf = property.Value.GetInt32();
+                    continue;
+                }
+            }
+            return new CompletionsOptions(Optional.ToList(prompt), Optional.ToNullable(maxTokens), Optional.ToNullable(temperature), Optional.ToNullable(topP), Optional.ToDictionary(logitBias), user, Optional.ToNullable(n), Optional.ToNullable(logprobs), Optional.ToNullable(echo), Optional.ToList(stop), Optional.ToNullable(presencePenalty), Optional.ToNullable(frequencyPenalty), Optional.ToNullable(bestOf));
+        }
+
+        /// <summary> Deserializes the model from a raw response. </summary>
+        /// <param name="response"> The response to deserialize the model from. </param>
+        internal static CompletionsOptions FromResponse(Response response)
```

Comment:
I don't think these are custom are they?  If not lets remove them from here and allow the generator to make them.

Reply:
Agreed! I moved all of the relevant `FromResponse` and `ToRequestContent` back into generated counterparts -- including a couple of proactive new ones for models not yet backed by typespec.